examples/microwave-oven-app/microwave-oven-common/src/microwave-oven-device.cpp

```diff
@@ -35,38 +35,79 @@ void ExampleMicrowaveOvenDevice::MicrowaveOvenInit()
     mOperationalStateInstance.Init();
     mMicrowaveOvenModeInstance.Init();
     mMicrowaveOvenControlInstance.Init();
+
+    // set default value for attribute SelectedWattIndex and WattRating
+    if (mMicrowaveOvenControlInstance.HasFeature(MicrowaveOvenControl::Feature::kPowerInWatts))
+    {
+        static_assert(ArraySize(mWattSettingList) > 0, "Watt setting list is empty!");
+        mSelectedWattIndex = ArraySize(mWattSettingList) - 1;
+        mWattRating        = mWattSettingList[mSelectedWattIndex];
+    }
+    else
+    {
+        mWattRating = kExampleWatt5;
+    }
 }
 
 /**
  * MicrowaveOvenControl cluster
  */
 Protocols::InteractionModel::Status
-ExampleMicrowaveOvenDevice::HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime, uint8_t powerSetting)
+ExampleMicrowaveOvenDevice::HandleSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTimeSec, bool startAfterSetting,
+                                                               Optional<uint8_t> powerSettingNum,
+                                                               Optional<uint8_t> wattSettingIndex)
 {
     // placeholder implementation
     Status status;
+    // update cook mode
     if ((status = mMicrowaveOvenModeInstance.UpdateCurrentMode(cookMode)) != Status::Success)
     {
         return status;
     }
-    mMicrowaveOvenControlInstance.SetCookTime(cookTime);
-    mMicrowaveOvenControlInstance.SetPowerSetting(powerSetting);
+
+    mMicrowaveOvenControlInstance.SetCookTimeSec(cookTimeSec);
+
+    // if using power as number, check if powerSettingNum has value before setting the power number
+    if (powerSettingNum.HasValue())
+    {
+        mPowerSettingNum = powerSettingNum.Value();
+    }
+
+    // if using power in watt, check if wattSettingIndex has value before setting the watt rating and watt list index
+    if (wattSettingIndex.HasValue())
+    {
+        mSelectedWattIndex = wattSettingIndex.Value();
+        mWattRating        = mWattSettingList[mSelectedWattIndex];
+    }
```

Comment:
Similar question here: spec says `[8.142] If this field is missing, the SelectedWattIndex attribute SHALL be set by the server to the index associated with the highest Watt setting for the selected CookMode.` ... where is this? Comment explaining this or operation should happen here.